src/view/View.js

```diff
@@ -0,0 +1,108 @@
+const { GAME_CONDITION } = require("../utils/Constants");
+const InputView = require("./InputView");
+const OutputView = require("./OutputView");
+const BridgeGameController = require("../BridgeGameController");
+const BridgeSizeValidator = require("../validator/BridgeSizeValidator");
+const BridgePositionValidator = require("../validator/BridgePositionValidator");
+const BridgeCommandValidator = require("../validator/BridgeCommandValidator");
+
+class View {
+  #gameController;
+
+  constructor() {
+    this.#gameController = new BridgeGameController();
+  }
+
+  startGame() {
+    OutputView.startGame();
+    this.#inputBridgeSize();
+  }
+
+  #inputBridgeSize() {
+    InputView.readBridgeSize(this.#getBridgeSize());
+  }
+
+  #getBridgeSize() {
+    return (size) => this.#handleSizeError(size);
+  }
+
+  #handleSizeError(size) {
+    const bridgeSizeValidator = new BridgeSizeValidator();
+
+    try {
+      bridgeSizeValidator.validate(size);
+      this.#gameController.inputBridgeSize(size);
+      this.#inputPositionToMove();
+    } catch (error) {
+      OutputView.printError(error.message);
+      this.#inputBridgeSize();
+    }
+  }
+
+  #inputPositionToMove() {
+    InputView.readMoving(this.#getPositionToMove());
+  }
+
+  #getPositionToMove() {
+    return (direction) => this.#handlePositionError(direction);
+  }
+
+  #handlePositionError(direction) {
+    const bridgePositionValidator = new BridgePositionValidator();
+    try {
+      bridgePositionValidator.validate(direction);
+      const isMove = this.#gameController.inputMove(direction);
+      this.#printBridge(this.#gameController.getMap());
+      this.#nextStep(isMove);
+    } catch (error) {
+      OutputView.printError(error.message);
+      this.#inputPositionToMove();
+    }
+  }
+
+  #nextStep(isMove) {
+    if (isMove && !this.#gameController.checkGameEnd()) {
+      return this.#inputPositionToMove();
+    }
+    if (!isMove) return this.#inputRetryOrQuit();
+    this.#quitGame();
+  }
```

Comment:
```suggestion
   #nextStep(isMove) {
    if (isMove) return this.#allPassOrNot();
    return this.#inputRetryOrQuit();
  }

  #allPassOrNot() {
    if (this.#gameController.checkGameEnd()) return this.#quitGame();
    return this.#inputPositionToMove();
  }
```
이 부분은 사실상 3가지 경우의 수가 있어 직관성이 조금 부족한 것 같습니다. 
함수 분리를 통해 더 간단히 만드는 방법도 있으니 참고 부탁드립니다.